### PR TITLE
STORM-853 Fix upload API to handle multi-args properly

### PR DESCRIPTION
**Pull request by HeartSaVioR**
* flatten arguments so that it can be treated to each arguments

Please refer https://issues.apache.org/jira/browse/STORM-853 for more details.

Please review and comment. Thanks in advance!

### CODE REVIEW

File: storm-core/src/clj/backtype/storm/ui/core.clj [hunk 1]
```diff
@@ -515,8 +515,7 @@
 
 (defn run-tplg-submit-cmd [tplg-jar-file tplg-config user]
   (let [tplg-main-class (if (not-nil? tplg-config) (trim (tplg-config "topologyMainClass")))
-        tplg-main-class-args (if (not-nil? tplg-config) (clojure.string/join " " (tplg-config "topologyMainClassArgs")))
-        tplg-jvm-opts (if (not-nil? tplg-config) (clojure.string/join " " (tplg-config "topologyJvmOpts")))
```
Comment (HeartSaVioR) [inline storm-core/src/clj/backtype/storm/ui/core.clj]: Removed unused variable.

**Comment by HeartSaVioR** [comment]
Seems like no argument makes another issue, fixing now.

**Comment by HeartSaVioR** [comment]
Case 1. 

> curl -i -b ~/cookiejar.txt -c ~/cookiejar.txt -X POST -F topologyConfig='{"topologyMainClass": "storm.starter.WordCountTopology"}' -F topologyJar=@storm-starter-topologies-0.10.0.jar http://localhost:8080/api/v1/uploadTopology

WordCountTopology runs via local mode -- OK

Case 2.

> curl -i -b ~/cookiejar.txt -c ~/cookiejar.txt -X POST -F topologyConfig='{"topologyMainClass": "storm.starter.WordCountTopology", "topologyMainClassArgs": []}' -F topologyJar=@storm-starter-topologies-0.10.0.jar http://localhost:8080/api/v1/uploadTopology

WordCountTopology runs via local mode -- OK

Case 3.

> curl -i -b ~/cookiejar.txt -c ~/cookiejar.txt -X POST -F topologyConfig='{"topologyMainClass": "storm.starter.WordCountTopology", "topologyMainClassArgs": ["a"]}' -F topologyJar=@storm-starter-topologies-0.10.0.jar http://localhost:8080/api/v1/uploadTopology

WordCountTopology runs via remote mode, which topology name is "a" -- OK

Case 4.

> curl  -i -b ~/cookiejar.txt -c ~/cookiejar.txt -X POST -F topologyConfig='{"topologyMainClass": "storm.starter.RollingTopWords", "topologyMainClassArgs": ["production-topology", "remote"]}' -F topologyJar=@storm-starter-topologies-0.10.0.jar http://localhost:8080/api/v1/uploadTopology

RollingTopWords runs via remote mode, which topology name is "production-topology" -- OK

**Comment by HeartSaVioR** [comment]
Note: When topology is run via local mode via uploadTopology REST API, UI server doesn't respond at that time, and send response when topology is finished or http request is canceled (Ctrl+C while waiting from curl).

**Comment by harshach** [comment]
+1.

**Comment by kishorvpatil** [comment]
+1

**Comment by ptgoetz** [comment]
+1 Thanks for the quick turnaround @HeartSaVioR.

**Pull request merged**